app/src/main/java/com/byoutline/kickmaterial/features/projectlist/ProjectListViewModel.kt

```diff
@@ -26,8 +26,10 @@ import javax.inject.Provider
 /**
  * @author Sebastian Kacprzak <sebastian.kacprzak at byoutline.com>
  */
-class ProjectListViewModel(showHeader: Boolean,
-                           val discoverField: ObservableCachedFieldWithArg<DiscoverResponse, DiscoverQuery>) {
+class ProjectListViewModel(
+        showHeader: Boolean,
+        val discoverField: ObservableCachedFieldWithArg<DiscoverResponse, DiscoverQuery>
+) : AttachableViewModel<ProjectClickListener>() {
```

Comment:
If viewModel is supposed to be attached and detached from view it should inherit from `AttachableViewModel`. 
This will allow `autoLifecycle` functions to call onAttach/onDetach viewModel functions automatically without needing to add boilerplate to Fragment/Activity